src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -273,7 +312,7 @@ private void AddTrivia(SyntaxNode node)
                 else if ((_options & SyntaxRemoveOptions.KeepEndOfLine) != 0
                     && HasEndOfLine(node.GetLeadingTrivia()))
                 {
-                    this.AddEndOfLine();
+                    this.AddEndOfLine(node.GetLeadingTrivia());
```

Comment:
it seems inefficient to call this stuff  twice.  why not have a function that gets the leading EndOfLine.  If you have one, then you add it.

Reply:
Done.